contracts/sol6/reserves/KyberFprReserveV2.sol

```diff
@@ -0,0 +1,349 @@
+pragma solidity 0.6.6;
+
+import "./IConversionRates.sol";
+import "./IWeth.sol";
+import "../IKyberSanity.sol";
+import "../IKyberReserve.sol";
+import "../IERC20.sol";
+import "../utils/Utils5.sol";
+import "../utils/Withdrawable3.sol";
+import "../utils/zeppelin/SafeERC20.sol";
+
+/// @title KyberFprReserve version 2
+/// Allow Reserve to work with both WETH by specifying address to hold WETH or ETH
+/// Allow Reserve to set maxGasPriceWei to trade with
+contract KyberFprReserveV2 is IKyberReserve, Utils5, Withdrawable3 {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    mapping(bytes32 => bool) public approvedWithdrawAddresses; // sha3(token,address)=>bool
```

Comment:
Small thing: Operationally, it's a little bit difficult to do keccak256(token,address) in order to find out whether an address has been whitelisted for approval. Not something that has to be fixed now, cos vv few project teams ask us about it. But it's good if there's a more convenient way.

Reply:
there is a getter function below: isAddressApprovedForWithdrawal
is that enough?
or other ideaS?

Reply:
Ah yeah, that will suffice. None that I can think of for now... @ayobuenavista @ssreelekhya any other feedback from ur interactions with the teams you've onboarded regarding reserves?